apps/examples/rtc/rtc_main.c

```diff
@@ -0,0 +1,222 @@
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <errno.h>
+#include <pthread.h>
+#include <sched.h>
+#include <stdio.h>
+#include <string.h>
+#include <fcntl.h>
+#include <time.h>
+#include <sys/ioctl.h>
+#include <sys/types.h>
+#include <tinyara/rtc.h>
+
+#define TEST_TIME 300	/* 300 secs */
+#define TEST_STACKSIZE 4096
+
+static volatile bool rtctest_started;
+
+static void rtc_test_stop(void)
+{
+	/* Has the rtc accuracy test already started? */
+	if (rtctest_started) {
+		/* Stop the test. */
+		printf("Stopping, not stopped yet.\n");
+		rtctest_started = FALSE;
+	} else {
+		printf("There is no active rtc accuracy test.\n");
+	}
+}
+
+static void *rtc_test(void *arg)
+{
+	int ret;
+	int fd;
+	struct rtc_time prev_time;
+	struct rtc_time next_time;
+	struct tm prev_tm;
+	struct tm next_tm;
+	time_t prev;
+	time_t next;
+	int total_cnt = 0;
+
+	printf("=== RTC Accuracy Test ===\n");
+
+	/* This sample runs forever with printing the log repeatedly on every TEST_TIME secs until "rtc stop".
+	 * You can check and compare the time which is from Serial Log Tool like Teraterm and the printed log from line 140.
+	 */
+
+	memset(&prev_time, 0, sizeof(struct rtc_time));
+	memset(&next_time, 0, sizeof(struct rtc_time));
+	memset(&prev_tm, 0, sizeof(struct tm));
+	memset(&next_tm, 0, sizeof(struct tm));
+
+	fd = open("/dev/rtc0", O_RDWR);
+	if (fd < 0) {
+		printf("ERROR : Fail to open rtc.\n");
+		return NULL;
+	}
+
+	/* Initialize the RTC time as 0 */
+	ret = ioctl(fd, RTC_SET_TIME, (unsigned long)&prev_time);
+	if (ret < 0) {
+		printf("ERROR : Fail to set RTC init time.\n");
+		close(fd);
+		return NULL;
+	}
+
+	sched_lock();
+	while (rtctest_started) {
+		sleep(TEST_TIME);
+		total_cnt++;
+
+		ret = ioctl(fd, RTC_RD_TIME, (unsigned long)&next_time);
+		if (ret < 0) {
+			printf("ERROR : Fail to get from RTC.\n");
+			close(fd);
+			sched_unlock();
+			return NULL;
+		} else {
+			memcpy(&next_tm, (struct tm *)&next_time, sizeof(struct tm));
+			memcpy(&prev_tm, (struct tm *)&prev_time, sizeof(struct tm));
+
+			prev = mktime(&prev_tm);
+			next = mktime(&next_tm);
+
+			printf("[Total %dm passed] Time Difference : %ds\n", total_cnt * (TEST_TIME / 60), next - prev);
+
+			/* Update the current time into 'prev_time' for next comparison. */
+			memcpy(&prev_time, &next_time, sizeof(struct rtc_time));
+
+			prev_time = next_time;
+
+		}
+	}
+
+	close(fd);
+	sched_unlock();
+	printf("Rtc Accuracy Test Stopped well\n");
+	return NULL;
+}
+
+/****************************************************************************
+ * rtc_main
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int rtc_main(int argc, char *argv[])
+#endif
+{
+#ifndef CONFIG_DISABLE_SIGNALS
```

Comment:
Could you let me know why you cover with disable_signals config? not disable_pthread?